src/main/java/bio/terra/service/resourcemanagement/google/GoogleResourceDao.java

```diff
@@ -119,14 +134,32 @@ public GoogleBucketResource createAndLockBucket(GoogleBucketRequest bucketReques
                 .name(bucketRequest.getBucketName())
                 .resourceId(keyHolder.getId())
                 .flightId(flightId);
-        } else {
-            return null;
         }
+
+        // The bucket already exists - locked or not. Retrieve it, validate it, and return it.
+        List<GoogleBucketResource> bucketResourcesByName = retrieveBucketsBy("name", bucketName, String.class);
```

Comment:
Since we are locking the table and making all other transactions wait, my preference is to give up that lock as soon as possible. For that reason, I would prefer that the SELECT be done after the table lock is released.

Reply:
There is no table unlock. That would require running another transaction. I think the complexity is not worth it.